rpc/src/lib.rs

```diff
@@ -1,2 +1,192 @@
-mod rpc;
-pub use rpc::{fetch_prover_input, FetchProverInputRequest};
+use alloy::{
+    providers::Provider,
+    rpc::types::eth::{BlockId, Withdrawal},
+    transports::Transport,
+};
+use anyhow::Context as _;
+use evm_arithmetization::proof::{BlockHashes, BlockMetadata};
+use futures::{StreamExt as _, TryStreamExt as _};
+use itertools::{Either, Itertools as _};
+use prover::ProverInput;
+use serde::Deserialize;
+use serde_json::json;
+use trace_decoder::{
+    trace_protocol::{BlockTrace, BlockTraceTriePreImages, TxnInfo},
+    types::{BlockLevelData, OtherBlockData},
+};
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "snake_case")]
+#[allow(clippy::large_enum_variant)]
+enum JerigonTrace {
+    Result(TxnInfo),
+    BlockWitness(BlockTraceTriePreImages),
+}
+
+const BLOCK_WITHOUT_FULL_TRANSACTIONS: bool = false;
```

Comment:
I feel this needs some documenting. I had to trace through the fn calls to work out what was meant by this.